### DISCUSSION TITLE
startup script is executed prior to stellarium finishing its initialization

### DISCUSSION THREAD

**Discussion by beltoforion** [Stellarium/stellarium#491] [Q&A] [answered]
### Expected Behaviour
Stellarium should not execute the startup script before it is completely initialized. An observer location set by the startup script is overwritten by stellariums own initialization.

### Actual Behaviour
An observer location set in a startup script is overwritten by the observer location determined during the startup of stellarium. in my case the location is set to Jena (Germany) instead of the one provided by the script.

### Steps to reproduce
Save the following line to a script:

core.setObserverLocation(13.34277, 50.911944, 425, 1, "Freiberg", "Earth");

then invoke the script as a startup script using the --startup-script parameter. The location will set to the location determined by stellarium during startup, not the one provided by the script.

### System
* Stellarium version: 0.18.2 (https://github.com/Stellarium/stellarium/releases/download/v0.18.2/Stellarium-0.18.2-x86_64.AppImage)
* Operating system: Linux Mint

### Logfile 
[log.txt](https://github.com/Stellarium/stellarium/files/2443540/log.txt)

**Comment by alex-w** [comment]
Is it a joke? You are enable automatic set the location after loading and initialization Stellarium with ignore of the current temporary location. Why you are expected some other behaviour?

**Comment by gzotti** [comment]
Alex I understand this is confusing. Actually I agree with the OP this behaviour is most likely not intended. If you want to occasionally start with a script, it should start only when the rest of initialisation has completed.

**Comment by beltoforion** [comment]
@alex-w Wow, what a welcome. Sorry but I do not understand your reply. My use case is the unattended, scripted execution of stellarium with a startup script to create videos for a certain time and location. 

I was under the impression that a script is supposed to behave in exactly the same way regardless whether you execute it as a startup script or at runtime of stellarium. This is not the case. This is not a big problem either since I can easily work around it with a core.wait(5).

**Comment by gzotti** [comment, marked as answer]
@beltoforion Do you have "Standort automatisch aus Netzwerk ermitteln" ("Get location from network") active in your location panel? This is the only scenario I can reproduce:

- Stellarium loads stored location
- Stellarium initializes location lookup from IP (if configured to do so)
- Stellarium loads location data from the commandline
- Stellarium loads the stored landscape (which may set location)
- other initialisation
- Stellarium executes startup script
- Stellarium receives answer from IP location lookup (if configured to do so) and sets location.

So yes, in your case either disable IP-based location lookup (saves network load!) or keep your wait(5)

**Comment by beltoforion** [comment]
Yes, that is the case. 
I'm fine with the sleep. Thanks for investigating this issue.
